sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -116,6 +116,9 @@ trait DataFrame extends RDDApi[Row] {
   /** Prints the schema to the console in a nice tree format. */
   def printSchema(): Unit
 
+  /** Prints the plan to the console for debugging purpose. */
+  def printPlan(extended: Boolean): Unit
```

Comment:
I thought explain was a pretty good name actually :)

@marmbrus what do you think?